tests/tt_metal/tt_metal/llk/test_single_core_matmul_int8.cpp

```diff
@@ -0,0 +1,230 @@
+    int8_t min = -5, max = 5;
+    std::vector<int8_t> input_0 = generate_uniform_int8(tile_size, min, max, 0 /*seed*/);
+    std::vector<int8_t> input_1 = generate_uniform_int8(tile_size, min, max, 1 /*different seed*/);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      Golden Generation
+    ////////////////////////////////////////////////////////////////////////////
+    std::vector<int32_t> golden_output_int32(tile_size, 0);
+    std::vector<int8_t> golden_output(tile_size, 0);
+    // matmul
+    for (int x = 0; x < 32; x++) {
+        for (int y = 0; y < 32; y++) {
+            for (int z = 0; z < 32; z++) {
+                golden_output_int32[get_output_coordinate(x, y)] +=
+                    (int32_t)input_0[get_output_coordinate(z, y)] * (int32_t)input_1[get_output_coordinate(x, z)];
+            }
+        }
+    }
+    // clamping to int8
+    for (int i = 0; i < 1024; i++) {
+        if (golden_output_int32[i] > 127) {
+            golden_output_int32[i] = 127;
+        } else if (golden_output_int32[i] < -127) {
+            golden_output_int32[i] = -127;
+        }
+        golden_output[i] = static_cast<int8_t>(golden_output_int32[i]);
+    }
+    convert_to_sign_mag(golden_output);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      Compile and Execute Application
+    ////////////////////////////////////////////////////////////////////////////
+
+    convert_to_sign_mag(input_0);
+    tt_metal::detail::WriteToBuffer(input0_dram_buffer, input_0);
+    convert_to_sign_mag(input_1);
+    tt_metal::detail::WriteToBuffer(input1_dram_buffer, input_1);
+
+    tt_metal::SetRuntimeArgs(
+        program,
+        reader_kernel,
+        core,
+        {
+            (uint32_t)in0_dram_addr,
+            (uint32_t)0,  // in_0 dram bank id
+            (uint32_t)in1_dram_addr,
+            (uint32_t)0,
+            (uint32_t)1,  // num_tiles
+        });
+    tt_metal::SetRuntimeArgs(
+        program,
+        writer_kernel,
+        core,
+        {
+            (uint32_t)out_dram_addr,
+            (uint32_t)0,
+            (uint32_t)1,  // num_tiles
+        });
+
+    tt_metal::detail::LaunchProgram(device, program);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      Comparison Checking
+    ////////////////////////////////////////////////////////////////////////////
+    std::vector<int8_t> dest_buffer_data;
+    tt_metal::detail::ReadFromBuffer(output_dram_buffer, dest_buffer_data);
+    pass = dest_buffer_data == golden_output;
+
+    for (int i = 0; i < 1024; i++) {
+        if (dest_buffer_data[i] != golden_output[i]) {
+            std::cout << "element " << i << ": " << (uint32_t)dest_buffer_data[i] << " vs. "
+                      << (uint32_t)golden_output[i] << " \n";
```

Comment:
```suggestion
    for (size_t i = 0; i < tile_size; i++) {
        if (dest_buffer_data[i] != golden_output[i]) {
            std::cerr << "element " << i << ": " << (uint32_t)dest_buffer_data[i] << " vs. "
                      << (uint32_t)golden_output[i] << " \n";
```
Please use `std:cerr` as it flushes immediately, so if your program crashes, you'll see all error messages. `std::cout` is buffered and might lose messages. On top of that, `std::cerr` is for errors/diagnostics, `std::cout` is for normal output.